client/my-sites/checkout/composite-checkout/components/prepurchase-notices/index.jsx

```diff
@@ -72,35 +71,41 @@ const PrePurchaseNotices = () => {
 		return products.filter( ( p ) => ! p.expired );
 	} );
 
-	/**
-	 * The active product on the current site that overlaps/conflicts with the plan currently in the cart.
-	 */
-	const siteProductThatOverlapsCartPlan = useSelector( ( state ) => {
+	const siteProductThatOverlapsCartPlan = useSelector( () => {
 		const planSlugInCart = cartItemSlugs.find( isJetpackPlanSlug );
-
 		if ( ! planSlugInCart ) {
 			return null;
 		}
 
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_BACKUPS ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_BACKUPS )
-		) {
-			return currentSiteProducts.find( isJetpackBackup );
-		}
-
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_ANTISPAM ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_ANTISPAM )
-		) {
-			return currentSiteProducts.find( isJetpackAntiSpam );
-		}
-
-		if (
-			planHasFeature( planSlugInCart, WPCOM_FEATURES_SCAN ) &&
-			siteHasFeature( state, siteId, WPCOM_FEATURES_SCAN )
-		) {
-			return currentSiteProducts.find( isJetpackScan );
+		//function that compares getSiteProducts with getAllProductsForPlan and returns the product that is in both
+		const getMatchingProducts = ( siteProducts, planSlug ) => {
+			// Get all features and products for the plan in the cart
+			const planFeatures = getAllFeaturesForPlan( planSlug );
+			const planProducts = getAllProductsForPlan( planSlug );
+
+			// Combine the plan features and products into a single array
+			const planItems = [ ...planFeatures, ...planProducts ];
+
+			//get all possible social slugs
+			const socialProductInSite = currentSiteProducts.find( ( product ) => {
+				return JETPACK_SOCIAL_PRODUCTS.includes( product.productSlug );
+			} );
+
+			// Filter the site products to only include those in the plan items
+			const matchingProducts = siteProducts.filter( ( product ) =>
+				planItems.includes( product.productSlug )
+			);
+
+			//special handling for sites with any social product
+			if ( socialProductInSite ) {
+				matchingProducts.push( socialProductInSite );
```

Comment:
If I'm understanding things correctly, when a user owns Social and has Complete in the cart, `matchingProducts` should already include the Social product.

It seems here that whenever a site has Social or a plan that includes it, we'll show the notice even though it's not Social that's overlapping. I did a test with a site with both Social and Backup, tried to purchase Security, and saw the Social notice in the checkout page, though the overlap is Backup in that case.

Reply:
So as far as the first part goes, it's like that because social products may not be an exact match as the site could have social basic which would be overlapped by social advanced in complete.  There might be a more elegant way to handle that. 

I do see where the logic is broken and it will display social overlapping the Security plan, so I'll work on that.